File: sdk/c#/PicarioXPO.RenderAPI/Fluent/FluentXpoUrlFactory.cs
using System;

namespace PicarioXPO.RenderAPI.Fluent
{
    public class FluentXpoUrlFactory : IFluentXpoUrlFactory
    {
        public IFluentXpoUrlGenerator CreateFluentUrlGenerator(IXpoUrlGenerator generator, FluentXpoUrlType urlType)
        {
            return new FluentXpoUrlGenerator(generator, GetUrlRequest(urlType));
        }

        private static XpoUrlRequest GetUrlRequest(FluentXpoUrlType urltype)
        {
            switch (urltype)
            {
                case FluentXpoUrlType.Image:
                    return new XpoImageUrlRequest();
                case FluentXpoUrlType.Coordinates:
                    return new XpoCoordinatesUrlRequest();
                default:
                    throw new InvalidOperationException("Cannot determine correct URL type");
            }
        }
    }
}

File: sdk/js/PicarioXPO.RenderAPI.V2/UrlGenerators/FluentXpoUrlFactory.ts
class FluentXpoUrlFactory implements UrlGeneratorModule.IFluentXpoUrlFactory {
    
    createFluentUrlGenerator(generator: UrlGeneratorModule.IXpoUrlGenerator, urlType: UrlGeneratorModule.FluentXpoUrlType, canvasGenerator?: UrlGeneratorModule.IXpoCanvasGenerator)
    {
        return new FluentXpoUrlGenerator(generator, this.getUrlRequest(urlType), canvasGenerator);
    }

    getUrlRequest(urltype: UrlGeneratorModule.FluentXpoUrlType )
    {
        switch (urltype) {
            case UrlGeneratorModule.FluentXpoUrlType.Image:
                return new XpoImageUrlRequest();
            case UrlGeneratorModule.FluentXpoUrlType.Coordinates:
                return new XpoCoordinatesUrlRequest();
            default:
                throw("Cannot determine correct URL type");
        }
    }
}